apps/ledger-live-mobile/src/reducers/accounts.ts

```diff
@@ -391,6 +423,163 @@ export const hasLendEnabledAccountsSelector = createSelector(
     }),
 );
 
+function accountHasPositiveBalance(account: AccountLike) {
+  return Boolean(account.balance?.gt(0));
+}
+
+export const accountsWithPositiveBalanceCountSelector = createSelector(
+  accountsSelector,
+  acc => acc.filter(accountHasPositiveBalance).length,
+);
+
+/**
+ * Selector creators
+ */
+
+type AccountsLikeSelector = OutputSelector<
+  State,
+  AccountLike[],
+  (res: Account[], res2: string[]) => AccountLike[]
+>;
+
+function makeAccountsCountSelectors(accountsSelector: AccountsLikeSelector) {
+  return createSelector(accountsSelector, accounts => accounts.length);
+}
+
+function makeHasAccountsSelectors(accountsSelector: AccountsLikeSelector) {
+  return createSelector(accountsSelector, accounts => accounts.length > 0);
+}
+
+function makeAccountsWithPositiveBalanceCountSelector(
+  accountsSelector: AccountsLikeSelector,
+) {
+  return createSelector(
+    accountsSelector,
+    accounts => accounts.filter(accountHasPositiveBalance).length,
+  );
+}
+
+function makeHasAccountsWithPositiveBalanceSelector(
+  accountsSelector: AccountsLikeSelector,
+) {
+  return createSelector(
+    accountsSelector,
+    accounts => accounts.filter(accountHasPositiveBalance).length > 0,
+  );
+}
+
+/**
+ * TOKEN-ACCOUNTS SELECTORS
+ */
+
+export const tokenAccountsSelector = createSelector(
+  accountsSelector,
+  accounts =>
+    flattenAccounts(accounts).filter(acc => acc.type === "TokenAccount"),
+);
+
+export const tokenAccountsNotBlacklistedSelector = createSelector(
+  tokenAccountsSelector,
+  blacklistedTokenIdsSelector,
+  (accounts, blacklistedIds) =>
+    accounts.filter(
+      acc => !blacklistedIds.includes(getAccountCurrency(acc).id),
+    ),
+);
+
+export const tokenAccountsCountSelector = makeAccountsCountSelectors(
+  tokenAccountsSelector,
+);
+
+export const tokenAccountsNotBlacklistedCountSelector =
+  makeAccountsCountSelectors(tokenAccountsNotBlacklistedSelector);
+
+export const hasTokenAccountsSelector = makeHasAccountsSelectors(
+  tokenAccountsSelector,
+);
+
+export const hasTokenAccountsNotBlacklistedSelector = makeHasAccountsSelectors(
+  tokenAccountsNotBlacklistedSelector,
+);
+
+export const tokenAccountsWithPositiveBalanceCountSelector =
+  makeAccountsWithPositiveBalanceCountSelector(tokenAccountsSelector);
+
+export const tokenAccountsNotBlackListedWithPositiveBalanceCountSelector =
+  makeAccountsWithPositiveBalanceCountSelector(
+    tokenAccountsNotBlacklistedSelector,
+  );
+
+export const hasTokenAccountsWithPositiveBalanceSelector =
+  makeHasAccountsWithPositiveBalanceSelector(tokenAccountsSelector);
+
+export const hasTokenAccountsNotBlackListedWithPositiveBalanceSelector =
+  makeHasAccountsWithPositiveBalanceSelector(
+    tokenAccountsNotBlacklistedSelector,
+  );
+
+/**
+ * NON-TOKEN-ACCOUNTS SELECTORS
+ */
```

Comment:
just a proposal but could we create a `accounts` folder here and split these selectors into smallers files and imports them in `index.ts` ? this file start to get quite huge

Reply:
True, I would split at least a file with all the reducers and a file with all the selectors, I think I'll do that in a separate PR as this one already has a lot of micro changes to review (and doing that means changing the imports in many many files...)